dataland-frontend/src/components/general/AlignedActivitiesDataTable.vue

```diff
@@ -0,0 +1,322 @@
+     * @returns a human readable version
+     */
+    camelCaseToWords(target: string): string {
+      return target.replace(/([A-Z]+)/g, " $1").replace(/([A-Z][a-z])/g, " $1");
+    },
+    /**
+     * @param key Define the column's CSS class
+     * @returns CSS class name
+     */
+    columnCss(key: string): string {
+      switch (key) {
+        case "activity":
+          return "col-activity";
+        case "naceCodes":
+          return "col-nace-codes";
+        default:
+          return "";
+      }
+    },
+    /**
+     * @param columnDefinition column definition we check against
+     * @param additionalClasses (optional) any additional classes to be added
+     * @returns classes for specific columns
+     */
+    groupColumnCssClasses(columnDefinition: MainColumnDefinition, additionalClasses = ""): string {
+      if (columnDefinition.groupIndex === 0) return `first-group-column ${additionalClasses}`;
+      return additionalClasses;
+    },
+    /**
+     * @param key the item to lookup
+     * @returns the display version of the column header
+     */
+    humanizeHeaderName(key: string) {
+      return this.columnHeaders[this.kpiKeyOfTable][key];
+    },
+  },
+});
+
+/**
+ * @param activity targeted activity object
+ * @returns list of revenue data items
+ */
+function createRevenueGroupData(activity: EuTaxonomyAlignedActivity): ActivityFieldValueObject[] {
+  return [
+    {
+      activity: activity.activityName as Activity,
+      group: "_revenue",
+      field: "revenue",
+      content: `${activity.share?.absoluteShare?.amount ?? ""} ${activity.share?.absoluteShare?.currency ?? ""}`,
+    },
+    {
+      activity: activity.activityName as Activity,
+      group: "_revenue",
+      field: "revenuePercent",
+      content: `${activity.share?.relativeShareInPercent ?? ""}%`,
+    },
+  ];
+}
+
+/**
+ * @param activity targeted activity object
+ * @returns list of substantial contribution criteria data items
+ */
+function createSubstantialContributionCriteriaGroupData(
+  activity: EuTaxonomyAlignedActivity,
+): ActivityFieldValueObject[] {
+  const fields = Object.entries(activity.substantialContributionCriteria ?? {});
+  return fields.map(([field, value]) => {
+    const content = value ? `${value}%` : "";
+    return {
+      activity: activity.activityName as string,
+      group: "substantialContributionCriteria",
+      field,
+      content,
+    };
+  });
+}
+
+/**
+ * @param activity targeted activity object
+ * @returns list of DNSH criteria data items
+ */
+function createDnshCriteriaGroupData(activity: EuTaxonomyAlignedActivity): ActivityFieldValueObject[] {
+  const fields = Object.entries(activity.dnshCriteria ?? {});
+  return fields.map(([field, value]) => {
+    const content = value ? `${value}` : "";
+    return {
+      activity: activity.activityName as string,
+      group: "dnshCriteria",
+      field,
+      content,
+    };
+  });
+}
+
```

Comment:
Does sonar complain about code duplications in functions `createSubstantialContributionCriteriaGroupData` and `createDnshCriteriaGroupData`? I think it would not be too expensive to combine them via providing "substantialContributionCriteria" or "dnshCriteria" as string input.

Reply:
Done.